flake.nix

```diff
@@ -0,0 +1,79 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    fenix.url = "github:nix-community/fenix";
+    fenix.inputs.nixpkgs.follows = "nixpkgs";
+    devenv.url = "github:cachix/devenv";
+    flake-utils.url = "github:numtide/flake-utils";
+    crane = {
+      url = "github:ipetkov/crane";
+      inputs = {
+        flake-utils.follows = "flake-utils";
+        nixpkgs.follows = "nixpkgs";
+      };
+    };
+  };
+
+  outputs = { self, nixpkgs, devenv, fenix, flake-utils, crane, ... } @ inputs:
+    flake-utils.lib.eachSystem [ flake-utils.lib.system.x86_64-linux ]
+      (system:
+        let
+          pkgs = import nixpkgs { inherit system; };
+          toolchain = (fenix.packages.${system}.fromToolchainFile {
+            file = ./rust-toolchain.toml;
+            sha256 = "sha256-gdYqng0y9iHYzYPAdkC/ka3DRny3La/S5G8ASj0Ayyc=";
+          });
+          craneLib = crane.lib.${system}.overrideToolchain toolchain;
+
+          src = craneLib.cleanCargoSource (craneLib.path ./.);
+
+          env = {
+            OPENSSL_LIB_DIR = "${pkgs.lib.getLib pkgs.openssl}/lib";
```

Comment:
`getLib` seems redundant here, isn't it?

Reply:
Interestingly enough, getLib's docs use openssl as the example 😄. It seems that it indeed became redundant at some point.